### ISSUE TITLE
CA1508 misses dead code

### ISSUE THREAD

**Issue by geeknoid** [dotnet/roslyn-analyzers#6497]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508)

### Describe the improvement

CA1508 is designed to detect dead code, but its analysis is missing trivial cases. Consider the code below:

```
    public void Foo(int j, int i, bool b)
    {
        bool a = false;
        if (a)
        {
            // 1. not reported
            throw new Exception();
        }

        if (!a || b)
        {
            // 2. not reported
            throw new Exception();
        }

        if (i != 0)
        {
            return;
        }

        if (j != 0)
        {
            return;
        }

        if (i != j)
        {
            // 3. reported
            throw new Exception();
        }
    }
```

On its side, sonar analyzer rule [S2583](https://rules.sonarsource.com/csharp/RSPEC-2583) detects the first two cases, but not the third.

The above issue occurs using the .NET 7 toolchain.
